src/App.tsx

```diff
@@ -869,24 +856,12 @@ function App(): ReactElement {
                     children: (
                       <div className={styles.tabContent}>
                         <SettingsTab
-                          // TODO: Refactor all of this into a settings or configuration object
-                          outOfRangeDrawSettings={outOfRangeDrawSettings}
-                          outlierDrawSettings={outlierDrawSettings}
-                          showScaleBar={showScaleBar}
-                          showTimestamp={showTimestamp}
+                          config={config}
+                          updateConfig={updateConfig}
```

Comment:
Is this a very standard practice to pass both the whole top-level state var and its updater function down to child components?  This looks like some kind of pattern or antipattern but I don't know which it is :)

Reply:
My guess is the only guidance React people would give here boils down to "do your best to encapsulate," i.e. keep state as far up the component tree as it can be and don't give child components access to data they don't need. This state needs to be at the top-level so this component can also pass it to `CanvasWrapper`, and all the settings are relevant to `SettingsTab` because that's where settings get set. So I think we're good!